exchanges/stream/websocket.go

```diff
@@ -677,76 +680,70 @@ func (w *Websocket) CanUseAuthenticatedWebsocketForWrapper() bool {
 
 // SetWebsocketURL sets websocket URL and can refresh underlying connections
 func (w *Websocket) SetWebsocketURL(url string, auth, reconnect bool) error {
-	defaultVals := url == "" || url == config.WebsocketURLNonDefaultMessage
-	if auth {
-		if defaultVals {
+	if url == "" || url == config.WebsocketURLNonDefaultMessage {
+		if auth {
 			url = w.defaultURLAuth
+		} else {
+			url = w.defaultURL
 		}
+	}
 
-		err := checkWebsocketURL(url)
-		if err != nil {
-			return err
-		}
-		w.runningURLAuth = url
-
-		if w.verbose {
-			log.Debugf(log.WebsocketMgr,
-				"%s websocket: setting authenticated websocket URL: %s\n",
-				w.exchangeName,
-				url)
-		}
+	err := checkWebsocketURL(url)
+	if err != nil {
+		return err
+	}
 
-		if w.AuthConn != nil {
-			w.AuthConn.SetURL(url)
-		}
+	message := "unauthenticated"
+	var conn Connection
+	if auth {
+		message = "authenticated"
+		conn = w.AuthConn
 	} else {
-		if defaultVals {
-			url = w.defaultURL
-		}
-		err := checkWebsocketURL(url)
-		if err != nil {
-			return err
-		}
-		w.runningURL = url
-
-		if w.verbose {
-			log.Debugf(log.WebsocketMgr,
-				"%s websocket: setting unauthenticated websocket URL: %s\n",
-				w.exchangeName,
-				url)
-		}
+		conn = w.Conn
+	}
+	if conn != nil {
+		conn.SetURL(url)
+	}
 
-		if w.Conn != nil {
-			w.Conn.SetURL(url)
-		}
+	if w.verbose {
+		log.Debugf(log.WebsocketMgr, "%s websocket: setting %s websocket URL: %s\n", w.exchangeName, message, url)
 	}
 
 	if w.IsConnected() && reconnect {
-		log.Debugf(log.WebsocketMgr,
-			"%s websocket: flushing websocket connection to %s\n",
-			w.exchangeName,
-			url)
+		log.Debugf(log.WebsocketMgr, "%s websocket: flushing websocket connection to %s\n", w.exchangeName, url)
 		return w.Shutdown()
 	}
 	return nil
 }
 
 // GetWebsocketURL returns the running websocket URL
-func (w *Websocket) GetWebsocketURL() string {
-	return w.runningURL
+func (w *Websocket) GetWebsocketURL(auth bool) string {
```

Comment:
Two issues with this:
* I don't think we should go further down the "websocket is a websocket but maybe actually 2 and one of them is auth" route.
* I don't think there's any cleaner API here in passing a boolean than having 2 methods

So in both cases I vote for GetWebsocketAuthURL() which has convenience handling for AuthConn is nil.